### PR TITLE
Adding go coverage reporting

### PR DESCRIPTION
**Pull request by radoslawc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:
This add gocov and gocov-html for pretty reports as well as coverage delta reporting tool , both for https://github.com/nephio-project/nephio/issues/374
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: images/gotests/coverage.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+REPORTS_DIR=$(pwd)/reports/
+
+mkdir -p $REPORTS_DIR/
+
+for go_main in $GOMAIN_DIRS; do
+    pushd "$go_main" >/dev/null
+    REPORT_FILENAME=$(tr '/' '_' <<< $go_main)
+    gocov test ./... | gocov-html > $REPORTS_DIR/$REPORT_FILENAME.html
+    popd >/dev/null
+done
```
Comment (electrocucaracha) [inline images/gotests/coverage.sh:21]:
```suggestion
REPORTS_DIR=$(pwd)/reports/

mkdir -p $REPORTS_DIR/

for go_main in $(egrep -rl --null --include \*.go 'package\s+main\b' | xargs -0 -L 1 dirname); do
    pushd "$go_main" >/dev/null
    gocov test ./... | gocov-html > $REPORTS_DIR/${go_main//\//_}.html
    popd >/dev/null
done
```

**Review by gvbalaji — APPROVED** [review]
/approve

File: images/gotests/coverage.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# http://www.apache.org/licenses/LICENSE-2.0
+##############################################################################
+
+REPORTS_DIR=$(pwd)/reports/
+
+mkdir -p $REPORTS_DIR/
+
+for go_main in $(egrep -rl --null --include \*.go 'package\s+main\b' | xargs -0 -L 1 dirname); do
+    pushd "$go_main" >/dev/null
+    gocov test ./... | gocov-html > $REPORTS_DIR/${go_main//\//_}.html
```
Comment (efiacor) [inline images/gotests/coverage.sh:17]: Should we add logic to exclude files from coverage reports in the project MakeFiles or could we do it here?

Comment (radoslawc) [reply images/gotests/coverage.sh:17]: I'd say here and with some kind of config file committed to repo might be good place to start.

**Pull request merged**